Helper/Order.php

```diff
@@ -1215,7 +1227,13 @@ public function prepareQuote($immutableQuote, $transaction)
         $boltParentQuoteId = $immutableQuote->getBoltParentQuoteId();
         if ($boltParentQuoteId) {
             $quote = $this->cartHelper->getQuoteById($boltParentQuoteId);
-            $this->cartHelper->replicateQuoteData($immutableQuote, $quote);
+            if($quote) {
+                $this->cartHelper->replicateQuoteData($immutableQuote, $quote);
+            } else {
+                // if the parent quote is removed then we create order by the immutable quote
+                $quote = $immutableQuote;
```

Comment:
This can have side effects further down the road. In hooks, for instance, dispatchPostCheckoutEvents won't be called. But twisting it more would require changing the stable working code logic and potentially break something else. As I consider this a very rare edge case I am approving it as is. The less changes the better.